net/host.go

```diff
@@ -237,7 +238,7 @@ func readStreamMessage(stream libp2pnetwork.Stream, maxMessageSize uint32) (comm
 // nextStreamMessage returns a channel that will receive the next message from the stream.
 // if there is an error reading from the stream, the channel will be closed, thus
 // the received value will be nil.
-func nextStreamMessage(stream libp2pnetwork.Stream, maxMessageSize uint32) <-chan common.Message {
+func nextStreamMessage(stream libp2pnetwork.Stream, maxMessageSize uint32) <-chan common.Message { //nolint:unparam
```

Comment:
`maxMessageSize` is shadowing the package-local constant with the same name. You could give the new protocol its own constant and use it, then rename the existing constant from `maxMessageSize` to something like `swapMaxMessageSize`. That would remove the shadowing and the linter complaint that we should eliminate the parameter because we only ever pass in a constant.